satellite/satellitedb/overlaycache.go

```diff
@@ -379,6 +379,58 @@ func (cache *overlaycache) IsVetted(ctx context.Context, id storj.NodeID, criter
 	return true, nil
 }
 
+// KnownOffline filters a set of nodes to offline nodes
+func (cache *overlaycache) KnownOffline(ctx context.Context, nodeIds storj.NodeIDList) (offlineNodes storj.NodeIDList, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if len(nodeIds) == 0 {
+		return nil, Error.New("no ids provided")
+	}
+
+	// get offline nodes
+	var rows *sql.Rows
+	switch t := cache.db.Driver().(type) {
+	case *sqlite3.SQLiteDriver:
+		args := make([]interface{}, 0, len(nodeIds))
+		for i := range nodeIds {
+			args = append(args, nodeIds[i].Bytes())
+		}
+
+		rows, err = cache.db.Query(cache.db.Rebind(`
+			SELECT id FROM nodes
+			WHERE id IN (?`+strings.Repeat(", ?", len(nodeIds)-1)+`)
+			AND last_contact_success < last_contact_failure
```

Comment:
By definition an online node is one that:
1. Has been successfully contacted since the last contact failure
2. Has been successfully contacted with a specific time frame (the `overlay.NodeCriteria.OnlineWindow` field).

I don't see the second criterion reflected in this query.